test/e2e/12_sync.bats

```diff
@@ -45,7 +47,38 @@ function setup() {
   [ "$sync_tag_hash" = "$head_hash" ]
 }
 
+@test "Sync fails on duplicate resource" {
+  # Wait until flux deploys the workloads
+  poll_until_true 'workload podinfo' 'kubectl -n demo describe deployment/podinfo'
+
+  # Check the sync tag
+  git pull -f --tags
+  local sync_tag_hash
+  sync_tag_hash=$(git rev-list -n 1 flux)
+  head_hash=$(git rev-list -n 1 HEAD)
+  [ "$sync_tag_hash" = "$head_hash" ]
+
+  # Bump the image of podinfo, duplicate the resource definition (to cause a sync failure)
+  # and make sure the sync doesn't go through
+  sed -i'.bak' 's%stefanprodan/podinfo:2.1.0%stefanprodan/podinfo:3.1.5%' "${clone_dir}/workloads/podinfo-dep.yaml"
+  cp "${clone_dir}/workloads/podinfo-dep.yaml" "${clone_dir}/workloads/podinfo-dep-2.yaml"
+  git add "${clone_dir}/workloads/podinfo-dep-2.yaml"
+  git -c 'user.email=foo@bar.com' -c 'user.name=Foo' commit -am "Bump podinfo and duplicate it to cause an error"
+  git push
+  # Wait until we find the duplicate failure in the logs
+  poll_until_true "duplicate resource in Flux logs" "kubectl logs -n $FLUX_NAMESPACE -l name=flux | grep -q \"duplicate definition of 'demo:deployment/podinfo'\""
+  # Make sure that the version of podinfo wasn't bumped
+  local podinfo_image
+  podinfo_image=$(kubectl get pod -n demo -l app=podinfo -o"jsonpath={['items'][0]['spec']['containers'][0]['image']}")
+  [ "$podinfo_image" = "stefanprodan/podinfo:2.1.0" ]
```

Comment:
Same here, probably better to get the image after the first successful sync, and compare it to the version you now have in memory here.